config/initializers/opentelemetry.rb

```diff
@@ -57,6 +57,13 @@
     # OTEL_RUBY_INSTRUMENTATION_PG_ENABLED=false
     # OTEL_RUBY_INSTRUMENTATION_REDIS_ENABLED=false
     # 
-    c.use_all()
+    # ActiveRecord instrumentation disabled while investigating interaction
+    # with attr_encrypted and both gems patching of ActiveRecord#reload
+    c.use_all({ 'OpenTelemetry::Instrumentation::ActiveRecord' => { enabled: false } })
   end
+
+  # Create spans for some ActiveRecord activity (queries, but not callbacks) with a subscription
+  # to `sql.active_record` https://guides.rubyonrails.org/active_support_instrumentation.html#active-record
+  tracer = OpenTelemetry.tracer_provider.tracer('OpenTelemetry::Instrumentation::ActiveSupport')
```

Comment:
Did you mean this? 🙂

```suggestion
  tracer = OpenTelemetry.tracer_provider.tracer('OpenTelemetry::Instrumentation::ActiveRecord')
```

Reply:
Maybe?

To loop in other readers: the first parameter to `OpenTelemetry.tracer_provider#tracer` is a `name` that will appear in telemetry produced by this tracer as either `scope.name` (current OTel spec) or `library.name` (slightly older OTel spec). Historically, `library.name` was "the name of the instrumentation library that produced this bit of data", but with the change to `scope.name`, it's meant to semantically convey the "[logical unit of the application code with which the emitted telemetry can be associated](https://opentelemetry.io/docs/reference/specification/glossary/#instrumentation-scope)."

Basically, the decision is which is more useful: (a) knowing what instrumentation produced the data or (2) from which component/subcomponent of the application the data was produced. 

Reflecting, I picked `ActiveSupport` because that's the name of the actual instrumentation that will be producing spans with this tracer. That's probably because I'm an instrumentation library maintainer and is often the question I have about telemetry. But `ActiveRecord` is more useful to more people to know that these spans are from the ActiveRecord component of their Rails application.

But the spans aren't being produced by the library that identifies itself as `OpenTelemetry::Instrumentation::ActiveRecord`.

... Phew. Tread lightly when twiddling semantic conventions in your schemas folks.

Reply:
Gotcha. I just remembered that we both kept getting AR and AS mixed up while we were pairing on this and wondered if this was another example of that. 😄

Reply:
> But the spans aren't being produced by the library that identifies itself as `OpenTelemetry::Instrumentation::ActiveRecord`.

I think the distinction here for me is that, technically, we used AS to instrument _but_ we were instrumenting AR. Teeeechnically we weren't using `OpenTelemetry::Instrumentation::ActiveSupport` to instrument, either, right?

I also agree that showing `ActiveRecord` would be far more intuitive when looking for spans about ActiveRecord operations, which is another thing that led me to comment here. :smile:

Reply:
🤔 

```suggestion
  tracer = OpenTelemetry.tracer_provider.tracer('ActiveRecord via ActiveSupport')
```

🤢

Reply:
```suggestion
  tracer = OpenTelemetry.tracer_provider.tracer("I'd just like to interject for a moment. The instrumentation you’re referring to as ActiveRecord, is in fact, ActiveSupport/ActiveRecord, or as I’ve recently taken to calling it, ActiveSupport plus ActiveRecord. ActiveRecord does not instrument itself, but rather is a component of a fully functioning instrumentation system made useful by `ActiveSupport::Notifications`, mixins and vital system components comprising a full instrumentation system as defined by OpenTelemetry")
```

Reply:
:shipit:

Reply:
```ruby
  # use a subscription to sql.active_record AS::Notification events as a substitute
  # for OTel ActiveRecord auto instrumentation
  ::OpenTelemetry::Instrumentation::ActiveSupport.subscribe(
    OpenTelemetry.tracer_provider.tracer('ActiveRecord'),
     'sql.active_record'
  )
```

Not a suggestion code block because I can't make it go from a one-line suggestion to multiple. 🤷🏻

Reply:
A'ight. That's [try it](https://github.com/mastodon/mastodon/pull/20338/commits/3716a14af3e929cd6d8a4f8bdb8800bc615d2d5b) this way.